### PR TITLE
feat: filemanager inventory

### LINKED ISSUE #219: filemanager: sync database to current S3 state
There should be functionality that allows the filemanager to sync it's database to the current S3 state. It is possible that the database state becomes out of sync, for example, due to events that fail to be ingested, or due to old objects that existed before the filemanager was deployed.

Some ways to accomplish this:
* Use the S3 Inventory.
    * This has the advantage of allowing someone to manually input an inventory report (i.e. call a filemanager API and sync with the given inventory report).
    * Could be configured to receive an inventory from a bucket (e.g. via [notifications](https://docs.aws.amazon.com/AmazonS3/latest/userguide/storage-inventory-notification.html)) on a daily/weekly basis?
    * Cannot be run on demand.
* Call S3 list operations on a bucket.
    * Could be run on demand.

### PR DESCRIPTION
**Pull request by mmalenic**
Closes #219 

### Changes
* Add support for ingesting events using [S3 inventory](https://docs.aws.amazon.com/AmazonS3/latest/userguide/storage-inventory.html) (or a custom inventory file uploaded to S3).
   * Events are parsed by fetching the `manifest.json` file and converting to database `FlatS3EventMessage` records.
   * Inventory events do not have sequencer values, so in order to bind to following deleted events, the sequencer is set to the empty string.
* Add integration tests using AWS generated inventory files.
   * Using some macros to include bytes and [`aws-smithy-mocks-experimental`](https://crates.io/crates/aws-smithy-mocks-experimental), which seems to work pretty well, and is more ergonomic than [mockall](https://crates.io/crates/mockall) and `#[double]`.
   
Note, I've removed Parquet and Orc from this PR so that it's less bulky. I'll will include it in the next one.

### Future
* I'd like to potentially try using some orm functionality from [sea-orm](https://github.com/SeaQL/sea-orm). Since sea-orm uses sqlx, we could opt in to using only the features that we want. For example, the [sea-orm-cli](https://github.com/SeaQL/sea-orm/tree/master/sea-orm-cli) `generate entity`, looks really useful, and could be added to `build.rs`. This inspects the database and generates strongly-typed Rust structs and enums for use in queries. Currently, using `FlatS3EventMessages` or other types is a bit bulky, especially in tests, where some files have more than 50% repetitive test data initialization code (although some well-placed macros would help with this). Let me know your thoughts @brainstorm.

### CODE REVIEW

File: lib/workload/stateless/stacks/filemanager/deploy/constructs/functions/function.ts [hunk 1]
```diff
@@ -190,6 +190,18 @@ export class Function extends Construct {
     this._role.addToPolicy(policyStatement)
   }
 
+  /**
```
Comment (brainstorm) [inline lib/workload/stateless/stacks/filemanager/deploy/constructs/functions/function.ts:193]: Nicely done! Totally stealing this function ;)

**Review by brainstorm — APPROVED** [review]
Glad to hear that the AWS experimental mock works well!

w.r.t to SeaORM, sure, you seem to be seeking for the migration features that EdgeDB has. I guess that there's 0 performance harm in having that CLI tooling that eases dev tasks, so go for it!

Kudos for this PR, learned a few things as usual, thank you!

**Comment by victorskl** [comment]
Can I review this on Monday first thing, pls. Need run for kid duty soon.

**Comment by victorskl** [comment]
+1 for sea-orm

**Comment by mmalenic** [comment]
> Can I review this on Monday first thing, pls. Need run for kid duty soon.

Sure, no rush!

**Review by victorskl — APPROVED** [review]
LGTM

**Pull request merged**